src/query/functions-v2/src/scalars/comparison.rs

```diff
@@ -0,0 +1,391 @@
+                );
+            }
+            DataType::Boolean => {
+                registry.register_2_arg::<BooleanType, BooleanType, BooleanType, _, _>(
+                    "lte",
+                    FunctionProperty::default(),
+                    |_, _| None,
+                    |lhs, rhs| (!lhs & rhs) || (lhs & rhs),
+                );
+            }
+            DataType::Number(u) => {
+                with_number_mapped_type!(|NUM_TYPE| match u {
+                    NumberDataType::NUM_TYPE => {
+                        registry
+                    .register_2_arg::<NumberType<NUM_TYPE>, NumberType<NUM_TYPE>, BooleanType, _, _>(
+                        "lte",
+                        FunctionProperty::default(),
+                        |_, _| None,
+                        |lhs, rhs| lhs <= rhs,
+                    );
+                    }
+                });
+            }
+            _ => todo!(),
+        });
+    }
+
+    // TODO: wait @b41sh complete jsonb cmp, and need support
+    // `VariantType like VariantType`
+    // `VariantType regexp VariantType`
+    // registry.register_2_arg::<VariantType, VariantType, BooleanType, _, _>(
+    //     "eq",
+    //     FunctionProperty::default(),
+    //     |_lhs, _rhs| None,
+    //     |lhs, rhs| lhs.cmp(rhs) == Ordering::Equal,
+    // );
+
+    // registry.register_2_arg::<VariantType, VariantType, BooleanType, _, _>(
+    //     "gt",
+    //     FunctionProperty::default(),
+    //     |_lhs, _rhs| None,
+    //     |lhs, rhs| lhs.cmp(rhs) == Ordering::Greater,
+    // );
+
+    // registry.register_2_arg::<VariantType, VariantType, BooleanType, _, _>(
+    //     "lt",
+    //     FunctionProperty::default(),
+    //     |_lhs, _rhs| None,
+    //     |lhs, rhs| lhs.cmp(rhs) == Ordering::Less,
+    // );
+
+    // registry.register_2_arg::<VariantType, VariantType, BooleanType, _, _>(
+    //     "gte",
+    //     FunctionProperty::default(),
+    //     |_lhs, _rhs| None,
+    //     |lhs, rhs| {
+    //         let res = lhs.cmp(rhs);
+    //         res == Ordering::Equal || res == Ordering::Greater
+    //     },
+    // );
+
+    // registry.register_2_arg::<VariantType, VariantType, BooleanType, _, _>(
+    //     "lte",
+    //     FunctionProperty::default(),
+    //     |_lhs, _rhs| None,
+    //     |lhs, rhs| {
+    //         let res = lhs.cmp(rhs);
+    //         res == Ordering::Equal || res == Ordering::Less
+    //     },
+    // );
+
+    // registry.register_2_arg::<VariantType, VariantType, BooleanType, _, _>(
+    //     "noteq",
+    //     FunctionProperty::default(),
+    //     |_lhs, _rhs| None,
+    //     |lhs, rhs| lhs.cmp(rhs) != Ordering::Equal,
+    // );
+
+    registry.register_2_arg::<StringType, StringType, BooleanType, _, _>(
+        "like",
+        FunctionProperty::default(),
+        |_lhs, _rhs| None,
+        |lhs, rhs| {
+            let pattern_str = simdutf8::basic::from_utf8(rhs)
```

Comment:
need to cache the pattern regexp, we can add todo comments there and improve it in the next pr.

Reply:
hmm, maybe we can create a hashmap to store the pattern?